src/cpu/pred/btb/pairtage.cc

```
@@ -0,0 +1,574 @@
+
+    assert(tableSizes.size() >= numPredictors);
+    assert(tableTagBits.size() >= numPredictors);
+    assert(histLengths.size() >= numPredictors);
+
+    tageTable.resize(numPredictors);
+    for (unsigned i = 0; i < numPredictors; ++i) {
+        tageTable[i].resize(tableSizes[i]);
+        for (unsigned j = 0; j < tableSizes[i]; ++j) {
+            tageTable[i][j].resize(numWays);
+        }
+        tableIndexBits[i] = ceilLog2(tableSizes[i]);
```

Cppcheck (2.20.0)

[error] Shifting 64-bit value by 64 bits is undefined behaviour

(shiftTooManyBits)